frontend/test/cypress/integration/product-list/filters.spec.ts

```diff
@@ -0,0 +1,167 @@
+const TEST_MIN_PRICE_INVALID = 100;
+const TEST_MIN_PRICE = 10;
+const TEST_MAX_PRICE = 50;
+
+describe('product list filters', () => {
+   const user = cy;
+   beforeEach(() => {
+      user.visit('/store/parts');
+   });
+
+   function getVirtualListClickOptions(): Partial<Cypress.ClickOptions> {
+      return {
+         // Cypress won't click the button because it detects that there's a parent
+         // element with the pointer-events: none style.
+         // Since this it's expected behavior for a virtual list while scrolling (
+         // https://github.com/bvaughn/react-window/issues/128#issuecomment-460163077 ),
+         // we need to force the click.
+         force: true,
+      };
+   }
+
+   function getVirtualListTypeOptions(): Partial<Cypress.TypeOptions> {
+      return {
+         // Cypress won't click the button because it detects that there's a parent
+         // element with the pointer-events: none style.
+         // Since this it's expected behavior for a virtual list while scrolling (
+         // https://github.com/bvaughn/react-window/issues/128#issuecomment-460163077 ),
+         // we need to force the click.
+         force: true,
+      };
+   }
+
+   function waitForSearchCompletion() {
+      user
+         .findByRole('region', { name: /products/i })
+         .as('section')
+         .should('have.attr', 'data-search-state', 'search');
+      return user
+         .get('@section')
+         .should('have.attr', 'data-search-state', 'idle');
+   }
+
+   it('should help user filter', () => {
```

Comment:
It seems like there are many different test cases in this single `it()`. Do you think we should split each test case into their own functions?

Reply:
Good point. I started that way, but then I've realized that the tests detailed [here](https://github.com/iFixit/react-commerce/issues/124) depend on each other (e.g.  `Click "Clear All"` assumes that the filters selected by the previous points are still there).
So either we keep it like this, or we change the tests so that they don't depend on each other.
I don't have strong opinions on this, so if you think the latter is better I can refactor

Reply:
I don't have super strong opinions either. I lean towards separate tests for each test case 🤷

Reply:
@mlahargou do you want me to do the refactor then?

Reply:
Works for me!